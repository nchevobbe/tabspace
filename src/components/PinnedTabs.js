// @flow
import type { TabType, TabIdType } from "../types.flow";
import React from "react";

type PropsType = {
  tabs: Array<TabType>,
  onTabClick: TabIdType => Promise<TabType>,
};

export default (props: PropsType) => {
  const { tabs, onTabClick } = props;
  if (!Array.isArray(tabs) || tabs.length === 0) {
    return null;
  }

  return (
    <aside className="pinned">
      {tabs.map(tab => (
        <a
          key={tab.id}
          href={tab.title}
          className="pinned-tab"
          onClick={() => onTabClick(tab.id)}
        >
          <img src={tab.favIconUrl} />
        </a>
      ))}
    </aside>
  );
};
