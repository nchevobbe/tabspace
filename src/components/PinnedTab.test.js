import React from "react";
import { shallow } from "enzyme";

import PinnedTabs from "./PinnedTabs";

describe("PinnedTabs", () => {
  it("renders empty", () => {
    expect(shallow(<PinnedTabs />)).toMatchSnapshot();
  });

  it("renders with tabs", () => {
    expect(
      shallow(
        <PinnedTabs
          tabs={[
            { id: 1, title: "tab-1", url: "url", favIconUrl: "/path/to/image" },
            { id: 2, title: "tab", url: "url", favIconUrl: "/path/to/image" },
            { id: 3, title: "tab", url: "url" },
          ]}
        />
      )
    ).toMatchSnapshot();
  });
});
