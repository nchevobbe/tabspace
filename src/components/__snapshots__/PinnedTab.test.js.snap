// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinnedTabs renders empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PinnedTabs renders with tabs 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    tabs={
      Array [
        Object {
          "favIconUrl": "/path/to/image",
          "id": 1,
          "title": "tab-1",
          "url": "url",
        },
        Object {
          "favIconUrl": "/path/to/image",
          "id": 2,
          "title": "tab",
          "url": "url",
        },
        Object {
          "id": 3,
          "title": "tab",
          "url": "url",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="pinned-tab"
          href="tab-1"
          onClick={[Function]}
        >
          <img
            src="/path/to/image"
          />
        </a>,
        <a
          className="pinned-tab"
          href="tab"
          onClick={[Function]}
        >
          <img
            src="/path/to/image"
          />
        </a>,
        <a
          className="pinned-tab"
          href="tab"
          onClick={[Function]}
        >
          <img
            src={undefined}
          />
        </a>,
      ],
      "className": "pinned",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <img
            src="/path/to/image"
          />,
          "className": "pinned-tab",
          "href": "tab-1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "/path/to/image",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <img
            src="/path/to/image"
          />,
          "className": "pinned-tab",
          "href": "tab",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "/path/to/image",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": <img
            src={undefined}
          />,
          "className": "pinned-tab",
          "href": "tab",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="pinned-tab"
            href="tab-1"
            onClick={[Function]}
          >
            <img
              src="/path/to/image"
            />
          </a>,
          <a
            className="pinned-tab"
            href="tab"
            onClick={[Function]}
          >
            <img
              src="/path/to/image"
            />
          </a>,
          <a
            className="pinned-tab"
            href="tab"
            onClick={[Function]}
          >
            <img
              src={undefined}
            />
          </a>,
        ],
        "className": "pinned",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="/path/to/image"
            />,
            "className": "pinned-tab",
            "href": "tab-1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "/path/to/image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="/path/to/image"
            />,
            "className": "pinned-tab",
            "href": "tab",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "/path/to/image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <img
              src={undefined}
            />,
            "className": "pinned-tab",
            "href": "tab",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
